### ISSUE TITLE
Install location doesn't appear to be honored

### ISSUE THREAD

**Issue by denelon** [vim/vim-win32-installer#219]
Passing /D to the installer to specify the install location doesn't appear to work correctly.

https://github.com/microsoft/winget-pkgs/issues/20228

**Comment by chrisbra** [comment]
I have a suspicion: 

https://github.com/vim/vim/blob/378697ac58b8f9705286e36d1805a052e96fb86e/nsis/gvim.nsi#L707-L716
```nsis
Function .onInit
!ifdef HAVE_MULTI_LANG
  # Select a language (or read from the registry).
  !insertmacro MUI_LANGDLL_DISPLAY
!endif

  # Check $VIM
  ReadEnvStr $INSTDIR "VIM"

  call CheckOldVim
```

It will set the INSTDIR variable to the content of the `$VIM` environment variable.  When this variable does not exists, the `$INSTDIR` variable will be empty. And this will overwrite the `/D=C:\vim` command line switch. Later the installer checks, that the variable is empty and uses the default. So I tried the following:

```
set VIM=c:\vim
gvim_8.2.3161_x64_signed.exe
```

It uses that directory. I am not sure this is actually useful (and seems to be un-documented). So how about to remove this ReadEnvStr part of the installer? (I played around with some functions, trying to only read the environment variable, if the $INSTDIR variable is not set to the default, but I think this does not make sense. 

I actually think, setting `$VIM` makes any sense, since Vim itself uses that variable to detect where it is installed. But if you set it to a directory that is not yet installed, and run a different vim (previous version) this may cause problems.

Here would be a patch: https://github.com/chrisbra/vim-win32-installer/blob/51cd3e7dba0182eee0619b5ebf294e7e9eec00a4/patch/nsis_installdir.patch

ping @k-takata @brammool

**Comment by k-takata** [comment]
I wasn't aware that the `/D` option is ignored.

> I actually think, setting `$VIM` doesn't make any sense,

Agree.

> Here would be a patch: https://github.com/chrisbra/vim-win32-installer/blob/master/patch/nsis_installdir.patch

Looks good.

**Comment by brammool** [comment]
Christian wrote:

> I have a suspicion: 
> 
> https://github.com/vim/vim/blob/378697ac58b8f9705286e36d1805a052e96fb86e/nsis/gvim.nsi#L707-L716
> ```nsis
> Function .onInit
> !ifdef HAVE_MULTI_LANG
>   # Select a language (or read from the registry).
>   !insertmacro MUI_LANGDLL_DISPLAY
> !endif
> 
>   # Check $VIM
>   ReadEnvStr $INSTDIR "VIM"
> 
>   call CheckOldVim
> ```
> 
> It will set the INSTDIR variable to the content of the `$VIM` environment variable.  When this variable does not exists, the `$INSTDIR` variable will be empty. And this will overwrite the `/D=C:\vim` command line switch. Later the installer checks, that the variable is empty and uses the default. So I tried the following:
> 
> ```
> set VIM=c:\vim
> gvim_8.2.3161_x64_signed.exe
> ```
> 
> It uses that directory. I am not sure this is actually useful (and seems to be un-documented). So how about to remove this ReadEnvStr part of the installer? (I played around with some functions, trying to only read the environment variable, if the $INSTDIR variable is not set to the default, but I think this does not make sense. 
> 
> I actually think, setting `$VIM` makes any sense, since Vim itself uses that variable to detect where it is installed. But if you set it to a directory that is not yet installed, and run a different vim (previous version) this may cause problems.
> 
> Here would be a patch: https://github.com/chrisbra/vim-win32-installer/blob/51cd3e7dba0182eee0619b5ebf294e7e9eec00a4/patch/nsis_installdir.patch
> 
> ping @k-takata @brammool 

I believe the original idea of using $VIM is that when Vim was
previously installed somewhere and $VIM would have been set to point to
it, installing a new version would put it in the same place.

The /D argument should overrule the value of $VIM.  How to do that I
don't know.

-- 
hundred-and-one symptoms of being an internet addict:
165. You have a web page burned into your glasses

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by brammool** [comment]
Should be fixed by patch 8.2.3214.
